tsl/src/bgw_policy/drop_chunks_api.c

```diff
@@ -57,27 +137,18 @@ drop_chunks_add_policy(PG_FUNCTION_ARGS)
 	Hypertable *hypertable;
 	Cache *hcache;
 	FormData_ts_interval *older_than;
+	DropChunksMeta meta;
+	Oid mapped_oid;
 	license_enforce_enterprise_enabled();
 	license_print_expiration_warning_if_needed();
 	ts_hypertable_permissions_check(ht_oid, GetUserId());
 
 	/* Make sure that an existing policy doesn't exist on this hypertable */
-	hypertable = ts_hypertable_cache_get_cache_and_entry(ht_oid, false, &hcache);
-
-	if (hypertable->fd.compressed)
-		ereport(ERROR,
-				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-				 errmsg("cannot add drop chunks policy to hypertable \"%s\" which contains "
-						"compressed data",
-						get_rel_name(ht_oid)),
-				 errhint("Please add the policy to the corresponding uncompressed hypertable "
-						 "instead.")));
-
-	older_than = ts_interval_from_sql_input(ht_oid,
-											older_than_datum,
-											older_than_type,
-											"older_than",
-											"add_drop_chunks_policy");
+	hcache = ts_hypertable_cache_pin();
```

Comment:
Do we need to unpin the cache after usage in this function?

Reply:
not needed as cache gets unpinned automatically at end of txn.

Reply:
Seems to be released already. 

Generally, though, probably good practice to have explicit unpins/releases to avoid holding on to memory longer than necessary.